src/EVM.hs

```diff
@@ -1321,7 +1321,7 @@ accessStorage addr slot continue = do
                 contract <- preuse (#cache % #fetched % ix addr')
                 case contract of
                   Nothing -> internalError "contract marked external not found in cache"
-                  Just fetched -> case readStorage (Lit slot') fetched.storage of
+                  Just fetched -> case readStorage (Expr.structureArraySlots (Lit slot')) fetched.storage of
```

Comment:
I guess these are here to make sure we only call it once? I'm not sure how I feel about having to remember to do this every time we call `readStorage`? Maybe we can just do it once at the start of `readStorage` and then pull all the actual logic out into some `go` in a where block that doesn't call `structureArraySlots`?

Reply:
Good point. I fixed this now, a lot cleaner!